### ISSUE TITLE
Short Phrase Workaround? location of specific words in seconds

### ISSUE THREAD

**Issue by sunnybala** [CorentinJ/Real-Time-Voice-Cloning#319]
So as other people have mentioned here, the spectrogram synthesis struggles with short phrases. One workaround is that if you lengthen your text to something that takes 6 seconds or so to say, you can do much better, then manually crop out the filler text.  However, has anyone had luck doing that in an automated way? (Without having to introduce a speech to text model...)

**Is there a way to locate the time in seconds that each character of the input string maps to?** That would make scrubbing out the filler text much easier.

 Example below.

input: **"this is cool"**
![image](https://user-images.githubusercontent.com/3474918/79070858-50309d00-7ca6-11ea-9efa-ee76be964c55.png)

This short phrase gives us a spectrogram that you might be able to tell will sound bad after going through the vocoder. The "washed out gap section" is a giveaway, and even the green sections lack the "grooves" that normal speech would have. After passing it through, none of the words are intelligible at all. 

Next, I'm just going to pad this with a bunch of the same word, dog. I timed this using a stopwatch and my own speaking to figure out how many times we needed to say "dog" to get to around 6 seconds. 

input: **"this is cool dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog dog"**
![image](https://user-images.githubusercontent.com/3474918/79070934-b5848e00-7ca6-11ea-9019-a96701a7d1c8.png)
Much better looking spectrogram and much better results as well. Every word sounds fine. At this point, all I need to do is remove the filler and I'll end up with just the piece I need.

![image](https://user-images.githubusercontent.com/3474918/79071235-a6064480-7ca8-11ea-87ae-1d62ddd45ca8.png)

Is it possible to extract the character locations in time of the synthesizer or anything? I'm using the code rather than the GUI so I can patch in wherever. Running a speech to text model and get word times that way, but it feels like overkill...but I'm open to any approach really. Just figured someone else had run into a similar issue before.

**Comment by haimin777** [comment]
Hello, for me works fine to cluster spectrogram and that delete all unnescessary parts
`import sklearn`
`signal = np.transpose(specs[0])`

`kmeans = sklearn.cluster.KMeans(n_clusters=2, random_state=0).fit(signal)`
`mask = kmeans.labels_.astype('bool')`
`results = np.transpose(signal[mask,...)`

**Comment by sunnybala** [comment]
that's a neat idea, thank you!

**Comment by AlioptK** [comment]
Hello, I tried the workaround that was proposed and it only works on a single short sentence (due to its two clusters).

I couldn't use it for my project because I have a random number of random sentences as input (I can't predict the number of clusters I'll need for each line).

Nevertheless, I managed to find another simple workaround that removes every pause. 
I'm pretty sure that it can still be improved, but it still works really well in my case.

In `/toolbox/__init__.py` after `spec = np.concatenate(specs, axis=1)` simply add this :
```
spec_temp = np.transpose(spec)
arr = np.empty((0,80)) 
threshold = -3.9 # -4.0 works too, but it's the absolute limit
for i in spec_temp:
  if  np.average(i)>= threshold:
    arr = np.append(arr, np.array([i]), axis=0)
spec = np.transpose(arr)
```

----

When I run the toolbox and ask to synthesize :
```
this is cool
My name is Emile
```
I get this (I added an intermediary mel spectrogram for you to see) : 

![mel_spectrograms](https://i.imgur.com/6qoZKb5.png)

No pause. But I can still get some strange sounds from time to time.
